frontend/packages/kubevirt-plugin/src/components/vms/vm-environment/selectors.ts

```diff
@@ -0,0 +1,131 @@
+import * as _ from 'lodash';
+import { sourcesEnum } from './types';
+import { getRandomChars } from '@console/dev-console/src/utils/shared-submit-utils';
+import { V1Volume } from '../../../types/vm/disk/V1Volume';
+
+export const getSourceName = (source): string => {
+  if (!source) {
+    return '';
+  }
+
+  return (
+    source.configMapRef?.name || source.secretRef?.name || source.serviceAccountRef?.name || ''
+  );
+};
+
+export const getSourceKind = (envDisk: (string | object | number)[]): string => {
+  // get only disk[1] as string
+  const ref = envDisk[1];
+  if (Object.keys(ref).includes('configMapRef')) {
```

Comment:
IMHO it may be nice to use `getEnvDiskRefKind` here ?

Reply:
done.